pkg/plugins/plugindef/plugindef.cue

```diff
@@ -34,6 +35,18 @@ seqs: [
 				// the UI.
 				name: string
 
+				// FIXME there appears to be a bug in thema that prevents this from working. Maybe it'd
+				// help to refer to it with an alias, but thema can't support using current list syntax.
+				// syntax (fixed by grafana/thema#82). Either way, for now, pascalName gets populated in Go.
+				let sani = (strings.ToTitle(regexp.ReplaceAllLiteral("[^a-zA-Z]+", name, "")))
+
+				// The PascalCase name for the plugin. Used for creating machine-friendly
+				// identifiers, typically in code generation.
+				//
+				// If not provided, defaults to name, but title-cased and sanitized (only
+				// alphabetical characters allowed).
+				pascalName: string & =~"^([A-Z][a-zA-Z]{1,62})$" | *sani
```

Comment:
I added this `pascalName` field to `plugindef` because without it, i was going in circles on how to align composable kinds with [the general naming system for kinds](https://github.com/grafana/grafana/blob/447e56b557310517f3e3fd1e0e1a5cac3d220bc5/pkg/kindsys/kindcats.cue#L29-L54). This value (along with id) is [injected](https://github.com/grafana/grafana/pull/61192/files#diff-2910f3982292a1758b4b3aca45ea9e7f032773018451d91e2f8f63323146ebaaR210-R217) into [the `grafanaplugin` cue instance](https://github.com/grafana/grafana/pull/61192/files#diff-f1d8e8d9498df64f5c6be3b0b25813d68a4891c14a78e10cac068d0a4b9823ebR12), and then used to construct a name for the composable kind. Thus, we end up with an identifier - used really for codegen, since composable kinds aren't ever a root resource - that looks like, say, 

To be clear, i'm not actually convinced we _have_ to make composable kinds follow the same naming patterns/that they make sense as for core and custom kinds. But this is the path of lesser resistance for now, and given that naming is hard, i think this is fine _for now_ and we can look at naming again in a follow-up.

There is some weird bug, i think in thema, which is causing the computed value (what's in `sani`) to not make it out when deserializing to a Go struct. So for now, there's [duplicate logic](https://github.com/grafana/grafana/pull/61192/files#diff-61f85a08bd5f398219965af5c1e4090ca2d24da0a5c2da35a70140693d9e31ddR43) to do this in Go (actually a bit more expressive, we can deal with it when the bug is fixed).

Reply:
Do we have a filed issue with the details? 🤔